### PR TITLE
Fix Zseri

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
So, I did a good chunk of it, but I'm stuck. I don't see how to easily get rid of the remaining gotos. The combo of 20/30/40  looks easy to remove, but the problem is there is a "goto 30" in the middle of the for loop below 40.

Is there an easy way to detangle them I'm missing @kortschak @sbinet ?

File: internal/amos/amos.go [hunk 1]
```diff
@@ -1946,234 +1956,159 @@ OneTen:
...
+	var I, IB, IDUM, IFLAG, IL, L, M, NN, NW int
+	var w [2]complex128
 	var sin, cos float64
+	var first bool
 
-	CONER = 1.0
-	NZ = 0
-	AZ = cmplx.Abs(complex(ZR, ZI))
-	if AZ == 0.0E0 {
-		goto OneSixty
```
Comment (kortschak) [inline internal/amos/amos.go:1964]: I'm confused about this going away?

Comment (kortschak) [inline internal/amos/amos.go:1964]: Ah, it's [here](https://github.com/gonum/mathext/pull/18/files#diff-f4a974e1c19635f3a4bd07971f170767R1983)

Comment (btracey) [reply internal/amos/amos.go:1964]: Yea, there's a bunch of re-arranging that's happening. There's still a bunch to go, but I got stuck, and making things around it simpler hasn't seemed to help.

**Comment by kortschak** [comment]
It can be done [this way](https://gist.github.com/kortschak/041df78c6e367a110e8642bb42611526). I don't think it should be. Probably this is just a case of finding better names for the labels and sucking up the gotos. I would suggest s/Twenty/loop/, s/Thirty/retry/ and s/Forty/done/.

**Comment by btracey** [comment]
PTAL

It isn't perfect, but I've spent way too much time on this already. I like this version better than the one with the gotos, because it makes the scope of certain variables much clearer. Maybe someone in the future can find a cleverer way to combine, or prove that some of the logic is redundant. (It feels like there is a way to condense all the `flag` logic)

File: internal/amos/amos.go [hunk 2]
```diff
@@ -1985,127 +1975,140 @@ func Zseri(z complex128, fnu float64, kode, n int, y []complex128, tol, elim, al
...
-			NW = Zuchk(s2, scale, tol)
-			if NW != 0 {
-				goto Thirty
+			s2 := s1 * coef
+			w[i] = s2
+			if flag == 1 {
+				if Zuchk(s2, scale, tol) != 0 {
+					var full bool
+					var dfnu float64
+					for {
```
Comment (kortschak) [inline internal/amos/amos.go]: Can you have a comment that this is essentially what exists above, and why.

**Review by kortschak — APPROVED** [review]

File: internal/amos/amos.go (hunk 2 above)
Comment (btracey) [reply internal/amos/amos.go]: Done.

**Pull request merged**